src/components/RoomNameInput.js

```diff
@@ -131,9 +146,13 @@ class RoomNameInput extends Component {
                 prefixCharacter="#"
                 placeholder={this.props.translate('newRoomPage.social')}
                 containerStyles={[styles.mb5]}
-                onChangeText={roomName => this.checkAndModifyRoomName(roomName)}
+                onChangeText={(roomName) => {
+                    const modifiedRoomName = this.modifyRoomName(roomName);
+                    this.setState({roomName: modifiedRoomName});
+                    this.validateRoomName(modifiedRoomName);
```

Comment:
Also, we can skip this function when `this.props.shouldShowErrorOnDemand` is false. 
```suggestion
		 if(!this.props.shouldShowErrorOnDemand){
		 	return;	
		 }
                 
                 this.validateRoomName(modifiedRoomName);
```